lib/services/livesync/android-livesync-library.ts

```diff
@@ -0,0 +1,396 @@
+			const socketId = this.socketConnection.uid;
+			const hash = crypto.createHash("md5").update(message).digest();
+
+			this.operationPromises[id] = {
+				resolve,
+				reject,
+				socketId
+			};
+
+			this.socketConnection.write(message);
+			this.socketConnection.write(hash);
+
+			setTimeout(() => {
+				if (this.isOperationInProgress(id)) {
+					this.handleSocketError(socketId, "Sync operation is taking too long");
+				}
+			}, 60000);
+		});
+
+		return operationPromise;
+	}
+
+	public end() {
+		if (this.socketConnection) {
+			this.socketConnection.end();
+		}
+	}
+
+	private sendFileHeader(filePath: string): Promise<void> {
+		return new Promise((resolve, reject) => {
+			let error;
+			this.verifyActiveConnection(reject);
+			const filePathData = this.getFilePathData(filePath);
+			const stats = this.$fs.getFsStats(filePathData.filePath);
+			const fileContentLengthBytes = stats.size;
+			const fileContentLengthString = fileContentLengthBytes.toString();
+			const fileContentLengthSize = Buffer.byteLength(fileContentLengthString);
+			const headerBuffer = Buffer.alloc(PROTOCOL_OPERATION_LENGTH_SIZE +
+					SIZE_BYTE_LENGTH +
+					filePathData.filePathLengthSize +
+					filePathData.filePathLengthBytes +
+					SIZE_BYTE_LENGTH +
+					fileContentLengthSize);
+
+			if (filePathData.filePathLengthSize > 255) {
+				error = this.getErrorWithMessage("File name size is longer that 255 digits.");
+			} else if (fileContentLengthSize > 255) {
+				error = this.getErrorWithMessage("File name size is longer that 255 digits.");
+			}
+
+			if (error) {
+				reject(error);
+			}
+
+			let offset = 0;
+			offset += headerBuffer.write(CREATE_FILE_OPERATION.toString(), offset, PROTOCOL_OPERATION_LENGTH_SIZE);
+			offset = headerBuffer.writeUInt8(filePathData.filePathLengthSize, offset);
+			offset += headerBuffer.write(filePathData.filePathLengthString, offset, filePathData.filePathLengthSize);
+			offset += headerBuffer.write(filePathData.relativeFilePath, offset, filePathData.filePathLengthBytes);
+			offset = headerBuffer.writeUInt8(fileContentLengthSize, offset);
+			headerBuffer.write(fileContentLengthString, offset, fileContentLengthSize);
+			const hash = crypto.createHash("md5").update(headerBuffer).digest();
+
+			this.socketConnection.write(headerBuffer);
+			this.socketConnection.write(hash);
+			resolve();
+		});
+	}
+
+	private sendFileContent(filePath: string): Promise<boolean> {
+		return new Promise((resolve, reject) => {
+			this.verifyActiveConnection(reject);
+			const fileStream = this.$fs.createReadStream(filePath);
+			const fileHash = crypto.createHash("md5");
+
+			fileStream
+				.on("data", (chunk: string | Buffer) => {
+					fileHash.update(chunk);
+					if (this.socketConnection) {
+						this.socketConnection.write(chunk);
+					} else {
+						const error = this.checkConnectionStatus();
+						//TODO Destroy method added in node 8.0.0.
+						//when we depricate node 6.x uncoment the line belolw
```

Comment:
Can't we use something like `process.version` and a `semver` check?

Reply:
Don't think its worth the effort, given we will make older node versions unsupported not that far in the future.